### ISSUE TITLE
Webkit settings do not get persisted in buffers

### ISSUE THREAD

**Issue by mianmoreno** [atlas-engineer/nyxt#2036]
**Describe the bug**
Often times, I like to navigate the web without media or images enabled by creating a mode similar to `no-image-mode`.

```lisp
(define-mode disable-media-mode ()
  "Disables media in the browser."
  ((constructor (lambda (mode)
                  (ffi-buffer-auto-load-image (buffer mode) nil)
                  (ffi-buffer-enable-media (buffer mode) nil)))   
   (visible-in-status-p nil)))
```

This works great for the most part. But I still get media when I access a media URL directly, such as opening an [image](https://nyxt.atlas.engineer/image/nyxt_128x128.png).

Further, when I delete the buffer and I restore it via `reopen-buffer`, the Webkit settings that were applied before are not anymore. I remember having seen an issue regarding buffer persistence somewhere in the project's issue tracker, but I can't recall which one it is and searching for it yielded no relevant results.

**Precise recipe to reproduce the issue**
Open a buffer with the aforementioned `disable-media-mode`, kill it, and re-open it via `reopen-buffer`, yielding the default Webkit settings and not the ones I specified.

For website-specific issues:
Can you reproduce this issue with Epiphany / GNOME Web (https://wiki.gnome.org/Apps/Web)?
No

**Information**
- OS name+version: guix (GNU Guix) ea11be50e9dbc5c043d3a18be8364b2e1b2507a5
- Graphics card and driver: Intel Corporation UHD Graphics
- Desktop environment / Window manager name+version: EXWM
- How you installed Nyxt (Guix pack, package manager, build from source): Guix pack
- Information from `show-system-information`:

```
Nyxt version: 2.1.1-1293-g45c399e4
Renderer version: GI-GTK
Operating system kernel: Linux 5.15.7
Lisp implementation: SBCL 2.1.9 (Dynamic space size: 3221225472)
Features: (:SERIES :SERIES-ANSI :SLYNK :WEBKIT2 :WEBKIT2-2.34 :WEBKIT2-CORS-ALLOWLIST
 :WEBKIT2-PASTE-PLAINTEXT :WEBKIT2-TRACKING :WEBKIT2-MUTE :WEBKIT2-EMOJI
 :WEBKIT2-MEDIA :WEBKIT2-SANDBOXING :GTK-3-22 :GTK-3-20 :GTK-3-18 :GTK-3-16
 :GTK-3-14 :GTK-3-12 :GTK-3-10 :GTK-3-8 :GTK-3-6 :GTK-3-4 :GTK :GDK-3-22
 :GDK-3-20 :GDK-3-18 :GDK-3-16 :GDK-3-14 :GDK-3-12 :GDK-3-10 :GDK-3-8 :GDK-3-6
 :GDK-3-4 :CAIRO-1-10 :CAIRO-1-12 :GDK-PIXBUF :GLIB-2-30 :GLIB-2-32 :GLIB-2-34
 :GLIB-2-36 :GLIB-2-38 :GLIB-2-40 :GLIB-2-42 :GLIB-2-44 :GLIB-2-46 :GLIB-2-48
 :GLIB-2-50 :GLIB-2-52 :GLIB-2-54 :GLIB-2-56 :GLIB-2-58 :GLIB :NYXT-2
 :FSET-EXT-STRINGS :CUSTOM-HASH-TABLE-NATIVE :SWANK :PLUMP-UTF-32 :GLOBAL-VARS
 :DECLARE-TYPES :PARENSCRIPT :SBCL+SAFE-STANDARD-READTABLE :NAMED-READTABLES
 :LPARALLEL :21BIT-CHARS :CHUNGA :FLEXI-STREAMS :CLOSER-MOP :CL-PPCRE-UNICODE
 :CL-UNICODE :CL-PPCRE :CL-JSON-DOUBLE-FLOAT-IS-SUBSUMED
 :CL-JSON-SINGLE-FLOAT-IS-SUBSUMED :BORDEAUX-THREADS :LPARALLEL.WITH-CLTL2
 :LPARALLEL.WITH-CAS :LPARALLEL.WITH-STEALING-SCHEDULER :SPLIT-SEQUENCE
 CHIPZ-SYSTEM:GRAY-STREAMS CFFI-FEATURES:FLAT-NAMESPACE CFFI-FEATURES:X86-64
 CFFI-FEATURES:UNIX :CFFI CFFI-SYS::FLAT-NAMESPACE ALEX::SEQUENCE-EMPTYP
 :FAST-IO-SV :FAST-IO :SBCL-USES-SB-ROTATE-BYTE :CL-JSON-CLOS :CL-JSON
 :THREAD-SUPPORT :ASDF3.3 :ASDF3.2 :ASDF3.1 :ASDF3 :ASDF2 :ASDF :OS-UNIX
 :NON-BASE-CHARS-EXIST-P :ASDF-UNICODE :X86-64 :GENCGC :64-BIT :ANSI-CL
 :COMMON-LISP :ELF :IEEE-FLOATING-POINT :LINUX :LITTLE-ENDIAN
 :PACKAGE-LOCAL-NICKNAMES :SB-CORE-COMPRESSION :SB-LDB :SB-PACKAGE-LOCKS
 :SB-THREAD :SB-UNICODE :SBCL :UNIX)

ASDF version: 3.3.5
ASDF registries: (NYXT-SOURCE-REGISTRY ENVIRONMENT-SOURCE-REGISTRY USER-SOURCE-REGISTRY
                      USER-SOURCE-REGISTRY-DIRECTORY
                      DEFAULT-USER-SOURCE-REGISTRY SYSTEM-SOURCE-REGISTRY
                      SYSTEM-SOURCE-REGISTRY-DIRECTORY
                      DEFAULT-SYSTEM-SOURCE-REGISTRY)
Critical dependencies: (/gnu/store/s7w144g61qsmschm6mzy41bvq0qvjfrs-cl-cffi-gtk-0.11.2-2.e9a46df/share/common-lisp/source/cl-cffi-gtk/gtk/cl-cffi-gtk.asd
 /gnu/store/w5awybp5s66bn7g74mbvlr10ivhr6m2v-cl-gobject-introspection-0.3-1.d0136c8/share/common-lisp/source/cl-gobject-introspection/cl-gobject-introspection.asd
 /gnu/store/gcqfm1frfrcdiizn6swqsy2znvyvmdqc-cl-webkit-3.5.0/share/common-lisp/source/cl-webkit/webkit2/cl-webkit2.asd)

Guix version: 
```

  If you can't run `show-system-information`, try `nyxt --system-information` from
  a shell.  It this still does not work, please provide the following:
  - Nyxt version (from the `nyxt-version` command or `nyxt --version` from a shell):
  - Lisp implementation/version (if built from source):
  - Kernel name+version:
  - WebKitGTK+ | QtWebEngine version:

**Output when started from a shell**

**Comment by aartaka** [comment]
You mean https://github.com/atlas-engineer/nyxt/issues/1095?

**Comment by efimerspan** [comment]
Not necessarily. In my case, I have said `disable-media-mode` in the `default-modes` slot for the buffer configuration and just override it on specific cases via `auto-mode`. So, no,  I believe this is not specific to `auto-mode`.

**Comment by aartaka** [comment]
Given that 
- #2627 fixes the last problem with buffer/mode state serialization and restoration,
- and Nyxt explicitly bears no responsibility for what's not bound as buffer/browser/profile/mode slots,
I'm marking this as a `wontfix` and linking it to #2627. To store WebKit settings and restore them gracefully, put those settings into modes, and use auto-rules/auto-mode with history (after #2627 is merged), where modes are saved.

**Comment by aadcg** [comment]
Won't fix means won't fix.
